### ISSUE TITLE
[macOS 14 Beta] Crash when use SDAnimatedImageRep for GIF on AppKit and macOS 14

### ISSUE THREAD

**Issue by dreampiggy** [SDWebImage/SDWebImage#3551]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. macos
 Platform Version        | e.g. 14.0 Developer Beta 1
 SDWebImage Version      | e.g. 5.12.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 14
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

Crash stacktrace:

![441686310362_ pic_back](https://github.com/SDWebImage/SDWebImage/assets/6919743/89c19b2f-3d06-4190-a8f7-863ec7fbc932)

Seems the danger pointer or CFRetain Count issue

**Comment by dreampiggy** [comment]
Resolved by Apple
